src/Core/Resolvers/CosmosMutationEngine.cs

```diff
@@ -94,17 +93,17 @@ private async Task<JObject> ExecuteAsync(IMiddlewareContext context, IDictionary
 
         /// <inheritdoc/>
         public void AuthorizeMutationFields(
+            string inputArgumentName,
             IMiddlewareContext context,
+            string clientRole,
             IDictionary<string, object?> parameters,
             string entityName,
             EntityActionOperation mutationOperation)
         {
-            string role = GetRoleOfGraphQLRequest(context);
```

Comment:
why remove this and then require it to be sent as an argument?

Reply:
The caller methods already had role evaluated, so I thought why evaluate again inside this method. But I think this is just creating an unnecessary dependency, since the role can be evaluated from context which is being passed anyways. I'll revert it back to previous state.